dds/src/implementation/dds_impl/user_defined_data_reader.rs

```diff
@@ -846,25 +847,26 @@ impl DdsShared<UserDefinedDataReader> {
         self.sample_rejected_status
             .write_lock()
             .increment(instance_handle, rejected_reason);
-        self.status_condition
-            .write_lock()
-            .add_communication_state(StatusKind::SampleRejected);
 
         if let Some(listener) = self.listener.write_lock().as_mut() {
             let status = self.get_sample_rejected_status();
             listener.trigger_on_sample_rejected(self, status);
         }
-    }
 
-    fn on_requested_deadline_missed(&self) {
         self.status_condition
             .write_lock()
-            .add_communication_state(StatusKind::RequestedDeadlineMissed);
+            .add_communication_state(StatusKind::SampleRejected);

```

Comment:
That does not match

Reply:
This is a heck of a bizarre change

Reply:
But it matches because line 859 is gone as well. This somehow belongs now to the function start at line 841. The new on_requested_deadline_missed is now at 861